onnxruntime/core/graph/model.h

```diff
@@ -296,6 +316,9 @@ class Model {
   // Path to model file. May be empty.
   const Path model_path_;
 
+  // Pointer to external data mapping table. May be nullptr.
+  const std::unordered_map<std::string, const void*>* external_data_map_;
```

Comment:
>onst std::unordered_map<std::string, const void*> [](http://example.com/codeflow?start=3&length=49)

Would it be possible to create an interface that can be re- implemented and deliver external data from different sources, not necessarily from the  map.
We can also add type safe data interfaces Data<T>() so the data ptr is already typed. We can also return typed data span gsl::span<const T>. The metadata should be provided by the API.

Reply:
Copying names into strings incurs memory allocation. Since the data passed in should be alive, then the strings should be as well. Can use std::string_view.
However, passing in OrtValues is a better option since it covers all kinds of data types.